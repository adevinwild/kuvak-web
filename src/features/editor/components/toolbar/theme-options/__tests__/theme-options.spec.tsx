import { render, screen } from '@testing-library/react'
import user from '@testing-library/user-event'
import ThemeOptions from '../'

function setupTests() {
  class MockPointerEvent extends Event {
    button: number
    ctrlKey: boolean
    pointerType: string

    constructor(type: string, props: PointerEventInit) {
      super(type, props)
      this.button = props.button || 0
      this.ctrlKey = props.ctrlKey || false
      this.pointerType = props.pointerType || 'mouse'
    }
  }

  window.PointerEvent = MockPointerEvent as unknown as typeof PointerEvent
  window.HTMLElement.prototype.scrollIntoView = jest.fn()
  window.HTMLElement.prototype.releasePointerCapture = jest.fn()
  window.HTMLElement.prototype.hasPointerCapture = jest.fn()
}

describe('ThemeOptions', () => {
  beforeAll(() => {
    setupTests()
  })

  it('ThemeOptions - should render correctly all the options', async () => {
    const { asFragment } = render(<ThemeOptions />)

    const select = screen.getByRole('combobox')

    const event = user.setup()
    await event.click(select)

    expect(asFragment()).toMatchSnapshot()
  })

  it("ThemeOptions - should have the default value of 'Light'", () => {
    render(<ThemeOptions />)

    expect(screen.getByRole('combobox')).toBeInTheDocument()
    expect(screen.getByRole('combobox')).toHaveTextContent('Light')
  })

  it('ThemeOptions - should change the theme to "Dark" when selected', async () => {
    render(<ThemeOptions />)
    const event = user.setup()

    const select = screen.getByRole('combobox')

    await event.click(select)
    expect(select).toHaveAttribute('aria-expanded', 'true')
    expect(screen.getByRole('listbox')).toBeInTheDocument()

    const firstOption = screen.getByText('Sunset')
    expect(firstOption).toBeInTheDocument()
    await event.click(firstOption)
    expect(select).toHaveTextContent('Sunset')
  })
})
