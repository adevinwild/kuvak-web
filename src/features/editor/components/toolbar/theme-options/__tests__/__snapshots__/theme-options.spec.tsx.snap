// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ThemeOptions ThemeOptions - should render correctly all the options 1`] = `
<DocumentFragment>
  <div
    class="grid gap-1.5 w-full"
  >
    <button
      aria-autocomplete="none"
      aria-controls="radix-:r0:"
      aria-expanded="true"
      class="flex h-10 items-center justify-between rounded-md border border-gray-200 bg-white px-3 py-2 text-sm ring-offset-white placeholder:text-gray-500 focus:outline-none focus:ring-2 focus:ring-gray-950 focus:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 [&>span]:line-clamp-1 dark:border-gray-800 dark:bg-gray-950 dark:ring-offset-gray-950 dark:placeholder:text-gray-400 dark:focus:ring-gray-300 w-24"
      data-state="open"
      data-testid="theme-options"
      dir="ltr"
      role="combobox"
      type="button"
    >
      <span
        style="pointer-events: none;"
      >
        Light
      </span>
      <svg
        aria-hidden="true"
        class="lucide lucide-chevron-down h-4 w-4 opacity-50"
        fill="none"
        height="24"
        stroke="currentColor"
        stroke-linecap="round"
        stroke-linejoin="round"
        stroke-width="2"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="m6 9 6 6 6-6"
        />
      </svg>
    </button>
  </div>
</DocumentFragment>
`;
